v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/source/reader/io/cassandra/iowrapper/CassandraDataSource.java

```diff
@@ -15,40 +15,77 @@
  */
 package com.google.cloud.teleport.v2.source.reader.io.cassandra.iowrapper;
 
+import com.datastax.oss.driver.api.core.ConsistencyLevel;
+import com.datastax.oss.driver.internal.core.retry.DefaultRetryPolicy;
 import com.google.auto.value.AutoValue;
 import com.google.common.collect.ImmutableList;
 import java.io.Serializable;
 import java.net.InetSocketAddress;
+import java.time.Duration;
 import java.util.List;
+import javax.annotation.Nullable;
 
 /**
  * Encapsulates details of a Cassandra Cluster. Cassandra Cluster can connect to multiple KeySpaces,
- * just like a Mysql instance can have multiple databases.
+ * just like a Mysql instance can have multiple databases. TODO(vardhanvthigle): Take
+ * DriverConfiguration as a GCS file for advanced overrides.
  */
 @AutoValue
 public abstract class CassandraDataSource implements Serializable {
 
   /** Name of the Cassandra Cluster. */
   public abstract String clusterName();
 
+  /** Name of local Datacenter. Must be specified if contactPoints are not empty */
+  @Nullable
+  public abstract String localDataCenter();
```

Comment:
Don't we also need to add auth details here? Like username password etc?

Reply:
Thanks. Added in the current push.